autofit/graphical/utils.py

```diff
@@ -54,6 +54,98 @@ def is_iterable(arg):
         arg, six.string_types
     )
 
+def nested_getitem(obj, key):
+    """
+    Example
+    -------
+    >>> nested_getitem([1, (2, 3), [3, {'a': 1, 'b': 2}]], (2, 1, 'b'))
+    2
+    """
+    for k in key:
+        obj = obj[k]
+    return obj 
+
+
+def nested_get(obj, key, default=None):
+    """
+    Example
+    -------
+    >>> nested_get([1, (2, 3), [3, {'a': 1, 'b': 2}]], (2, 1, 'b'))
+    2
+    >>> nested_get([1, (2, 3), [3, {'a': 1, 'b': 2}]], (2, 1, 'c'), 'default')
+    'default'
+    """
+    try:
+        return nested_getitem(obj, key)
+    except (KeyError, IndexError):
+        return default
+    
+
+def nested_set(obj, key, val):
+    """
+    Example
+    -------
+    >>> obj = [1, (2, 3), [3, {'a': 1, 'b': 2}]]
+    >>> nested_set(obj, (2, 1, 'b'), 3)
+    >>> obj
+    [1, (2, 3), [3, {'a': 1, 'b': 3}]]
+
+    >>> nested_set(obj, (2, 1,), ()))
+    >>> obj
+    [1, (2, 3), [3, ()]]
+    """
+    *parents, k = key 
+    for p in parents:
+        obj = obj[p]
+    obj[k] = val 
+
+
+def nested_items(*args, key=()):
+    """
+    Example
+    -------
+    >>> list(nested_items([1, (2, 3), [3, {'a': 1, 'b': 2}]]))
+    [((0,), 1), ((1, 0), 2), ((1, 1), 3), ((2, 0), 3), ((2, 1, 'a'), 1), ((2, 1, 'b'), 2)]
+
+    >>> list(nested_items([1, (2, 3)], [4, (5, 6)]))
+    [((0,), 1, 4), ((1, 0), 2, 5), ((1, 1), 3, 6)]
+    """
+    out, *_ = args
+    if isinstance(out, dict):
+        for k in out:
```

Comment:
to match `jax` functionality (and to provide stable ordering) may want `from k in sorted(out)` @rhayes777

Reply:
Makes sense